Lib/encodings/idna.py

```diff
@@ -102,23 +100,19 @@ def ToASCII(label):
     # Step 5: Check ACE prefix
     if label.startswith(sace_prefix):
         raise UnicodeEncodeError(
-            "idna", label.decode("ascii", errors="backslashreplace"),
-            0, len(sace_prefix), "Label starts with ACE prefix")
+            "idna", label, 0, len(sace_prefix), "Label starts with ACE prefix")
 
     # Step 6: Encode with PUNYCODE
-    label = label.encode("punycode")
+    label_ascii = label.encode("punycode")
 
     # Step 7: Prepend ACE prefix
-    label = ace_prefix + label
+    label_ascii = ace_prefix + label_ascii
 
     # Step 8: Check size
-    if 0 < len(label) < 64:
-        return label
-    label = label[len(ace_prefix):].decode("punycode", errors="replace")
-    if len(label) == 0:
-        raise UnicodeEncodeError("idna", label, 0, 1, "label empty")
-    else:
-        raise UnicodeEncodeError("idna", label, 0, len(label), "label too long")
+    # do not check for empty as we prepend ace_prefix.
+    if len(label_ascii) < 64:
+        return label_ascii
+    raise UnicodeEncodeError("idna", label, 0, len(label), "label too long")
```

Comment:
The reason I had a special case for `len(label) == 0` is because if you pass `0, 0` as the range, the error will print the inclusive range as `0--1`, instead of `0-0` for the empty case

Reply:
See "# Step 7: Prepend ACE prefix".
label must not be empty.